src/screens/smart-screen/smart-screen.tsx

```diff
@@ -0,0 +1,216 @@
+}
+
+const mapDispatchToProps = {
+    fetchScreenData,
+
+    claimRewardNoInput,
+    withdraw
+};
+
+interface IState {
+    screenData: IScreenData;
+    isLoading: boolean;
+    error: any;
+}
+
+export class SmartScreenComponent extends React.Component<
+    IReduxProps & IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>,
+    IState
+> {
+    constructor(
+        props: IReduxProps & IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>
+    ) {
+        super(props);
+        this.state = {
+            screenData: undefined,
+            isLoading: true,
+            error: undefined
+        };
+    }
+
+    public componentDidMount() {
+        this.props.fetchScreenData(this.props.context);
+    }
+
+    public componentDidUpdate(prevProps: IReduxProps & IExternalProps) {
+        if (
+            (this.props.account && this.props.account !== prevProps.account) ||
+            (this.props.walletPublicKey &&
+                this.props.walletPublicKey !== prevProps.walletPublicKey) ||
+            (this.props.chainId && this.props.chainId !== prevProps.chainId) ||
+            this.props.context?.tab !== prevProps.context?.tab
+        ) {
+            this.props.fetchScreenData(this.props.context);
+            this.getScreenData();
+        }
+
+        // TODO: needs to be fixed for tab account
+
+        if (
+            (this.props.dashboard && this.props.dashboard !== prevProps.dashboard) ||
+            (this.props.token && this.props.token !== prevProps.token)
+        ) {
+            this.getScreenData();
+        }
+    }
+
+    private getScreenData() {
+        const { dashboard, token } = this.props;
+
+        const screenKey = getScreenDataKey({
+            pubKey: this.props.walletPublicKey,
+            blockchain: this.props.account.blockchain,
+            chainId: this.props.chainId,
+            address: this.props.account.address,
+            tab: this.props.context?.tab
+        });
+
+        if (dashboard && dashboard[screenKey]) {
+            const screenData: IScreenData = dashboard[screenKey];
+
+            screenData &&
+                this.setState({
+                    screenData,
+                    isLoading: screenData.isLoading,
+                    error: screenData.error
+                });
+        } else if (token && token[screenKey]) {
+            const screenData: IScreenData = token[screenKey];
+
+            screenData &&
+                this.setState({
+                    screenData,
+                    isLoading: screenData.isLoading,
+                    error: screenData.error
+                });
+        } else {
+            this.setState({
+                screenData: undefined,
+                isLoading: true,
+                error: undefined
+            });
+        }
+    }
+
+    public render() {
+        const { styles, theme } = this.props;
+        const { screenData } = this.state;
+
+        if (this.state.isLoading) {
```

Comment:
you should display a skeleton only if you the state is loading and there is no data in screen data